core/register_core_types.cpp

```diff
@@ -141,42 +141,42 @@ void register_core_types() {
 
 	GDREGISTER_CLASS(Object);
 
-	GDREGISTER_VIRTUAL_CLASS(Script);
+	GDREGISTER_ABSTRACT_CLASS(Script);
 
 	GDREGISTER_CLASS(RefCounted);
 	GDREGISTER_CLASS(WeakRef);
 	GDREGISTER_CLASS(Resource);
 	GDREGISTER_CLASS(Image);
 
 	GDREGISTER_CLASS(Shortcut);
-	GDREGISTER_VIRTUAL_CLASS(InputEvent);
-	GDREGISTER_VIRTUAL_CLASS(InputEventWithModifiers);
-	GDREGISTER_VIRTUAL_CLASS(InputEventFromWindow);
+	GDREGISTER_ABSTRACT_CLASS(InputEvent);
+	GDREGISTER_ABSTRACT_CLASS(InputEventWithModifiers);
+	GDREGISTER_ABSTRACT_CLASS(InputEventFromWindow);
 	GDREGISTER_CLASS(InputEventKey);
 	GDREGISTER_CLASS(InputEventShortcut);
-	GDREGISTER_VIRTUAL_CLASS(InputEventMouse);
+	GDREGISTER_ABSTRACT_CLASS(InputEventMouse);
 	GDREGISTER_CLASS(InputEventMouseButton);
 	GDREGISTER_CLASS(InputEventMouseMotion);
 	GDREGISTER_CLASS(InputEventJoypadButton);
 	GDREGISTER_CLASS(InputEventJoypadMotion);
 	GDREGISTER_CLASS(InputEventScreenDrag);
 	GDREGISTER_CLASS(InputEventScreenTouch);
 	GDREGISTER_CLASS(InputEventAction);
-	GDREGISTER_VIRTUAL_CLASS(InputEventGesture);
+	GDREGISTER_ABSTRACT_CLASS(InputEventGesture);
 	GDREGISTER_CLASS(InputEventMagnifyGesture);
 	GDREGISTER_CLASS(InputEventPanGesture);
 	GDREGISTER_CLASS(InputEventMIDI);
 
 	// Network
-	GDREGISTER_VIRTUAL_CLASS(IP);
+	GDREGISTER_ABSTRACT_CLASS(IP);
 
-	GDREGISTER_VIRTUAL_CLASS(StreamPeer);
+	GDREGISTER_ABSTRACT_CLASS(StreamPeer);
 	GDREGISTER_CLASS(StreamPeerExtension);
```

Comment:
`StreamPeer`, `PacketPeer` and `MultiplayerPeer` all have an `*Extension` wrapper which AFAIU was specifically added to workaround the limitation that this PR lifts. So these could likely be registered as virtual instead, and the `*Extension` wrappers dropped.

But that's probably stuff for a future PR so it might be good to keep it as is in this one. CC @Faless

Reply:
Extension makes sense if its something very large, so I guess it is up to Fabio to change.